ash/src/extensions/khr/maintenance4.rs

```diff
@@ -0,0 +1,98 @@
+use crate::vk;
+use crate::{Device, Instance};
+use std::ffi::CStr;
+use std::mem;
+use std::ops::DerefMut;
+
+#[derive(Clone)]
+pub struct Maintenance4 {
+    handle: vk::Device,
+    fns: vk::KhrMaintenance4Fn,
+}
+
+impl Maintenance4 {
+    pub fn new(instance: &Instance, device: &Device) -> Self {
+        let fns = vk::KhrMaintenance4Fn::load(|name| unsafe {
+            mem::transmute(instance.get_device_proc_addr(device.handle(), name.as_ptr()))
+        });
+        Self {
+            handle: device.handle(),
+            fns,
+        }
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkGetDeviceBufferMemoryRequirementsKHR.html>"]
+    pub unsafe fn get_device_buffer_memory_requirements<'a>(
+        &self,
+        create_info: &vk::DeviceBufferMemoryRequirementsKHR,
+        next: Option<&'a mut impl vk::ExtendsMemoryRequirements2>,
+    ) -> vk::MemoryRequirements2Builder<'a> {
+        let mut memory_requirements = vk::MemoryRequirements2::builder();
+        if let Some(next) = next {
+            memory_requirements = memory_requirements.push_next(next);
+        }
+
+        self.fns.get_device_buffer_memory_requirements_khr(
+            self.handle,
+            create_info,
+            memory_requirements.deref_mut(),
+        );
+
+        memory_requirements
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkGetDeviceImageMemoryRequirementsKHR.html>"]
+    pub unsafe fn get_device_image_memory_requirements<'a>(
+        &self,
+        create_info: &vk::DeviceImageMemoryRequirementsKHR,
+        next: Option<&'a mut impl vk::ExtendsMemoryRequirements2>,
+    ) -> vk::MemoryRequirements2Builder<'a> {
+        let mut memory_requirements = vk::MemoryRequirements2::builder();
+        if let Some(next) = next {
+            memory_requirements = memory_requirements.push_next(next);
+        }
+
+        self.fns.get_device_image_memory_requirements_khr(
+            self.handle,
+            create_info,
+            memory_requirements.deref_mut(),
+        );
+
+        memory_requirements
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkGetDeviceImageSparseMemoryRequirementsKHR.html>"]
+    pub unsafe fn get_device_image_sparse_memory_requirements(
+        &self,
+        create_info: &vk::DeviceImageMemoryRequirementsKHR,
+    ) -> Vec<vk::SparseImageMemoryRequirements2> {
+        let mut count = 0;
+        self.fns.get_device_image_sparse_memory_requirements_khr(
+            self.handle,
+            create_info,
+            &mut count,
+            std::ptr::null_mut(),
+        );
+        let mut data = Vec::with_capacity(count as usize);
+        self.fns.get_device_image_sparse_memory_requirements_khr(
+            self.handle,
+            create_info,
+            &mut count,
+            data.as_mut_ptr(),
+        );
+        data.set_len(count as usize);
+        data
```

Comment:
This should probably be using `read_into_defaulted_vector`.

Reply:
That was omitted when quickly fixing up the wrong trait type - I want to split this back out anyway to be consistent with the other `_len()` functions.